### PR TITLE
Rust UI: keyboards tweaks and fixes

### LINKED ISSUE #2142: Keyboards for seed restore

### PR DESCRIPTION
**Pull request by mmilata**
Fixes #2142. Obsoletes #2104.

emu00000000 emu00000001 emu00000002 emu00000003 emu00000004 emu00000005 emu00000006 emu00000007 emu00000008 emu00000009

Missing:
* return entered sequence (needs #2138)
* fix for visual glitches caused by overlapping `Maybe`s (needs Honza's placement code)
* long PINs and passphrases
* show wrong pin warning (needs ability to start timer on layout init)
* funny button shape in PIN keyboard

### CODE REVIEW

**Comment by matejcik** [comment]
on image 1 the "disabled OK" button looks bad. IIRC the checkmark should be white (despite being disabled)

**Comment by matejcik** [comment]
please rebase (or maybe just cherry-pick?) onto current master

**Comment by prusnak** [comment]
> on image 1 the "disabled OK" button looks bad. IIRC the checkmark should be white (despite being disabled)
 
Shall we even show the Confirm button in case it is disabled?

File: core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs [hunk 1]
```diff
@@ -86,36 +123,50 @@ impl PinKeyboard {
...
+        let reset_cancel_area = grid.row_col(3, 0);
+        self.reset_btn.place(reset_cancel_area);
+        self.cancel_btn.place(reset_cancel_area);
+        self.confirm_btn.place(grid.row_col(3, 2));
 
         // Digit buttons.
         for (i, btn) in self.digit_btns.iter_mut().enumerate() {
             // Assign the digits to buttons on a 5x3 grid, starting from the second row.
             let area = grid.cell(if i < 9 {
                 // The grid has 3 columns, and we skip the first row.
```
Comment (matejcik) [inline core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs:165]: "and we skip first row" is no longer true

**Review by matejcik — COMMENTED** [review]
would it be possible to run UI2 ui tests as a separate CI job?

File: core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs (hunk 1 above)
Comment (mmilata) [reply core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs]: fixed in [e579140](https://github.com/trezor/trezor-firmware/pull/2154/commits/e5791405c4c1cb4f1738cfbffbb652f6cd691394)

**Review by matejcik — APPROVED** [review]

**Pull request merged**